### PR TITLE
scene length setting referenced from DB record for 3dsMax

### PR DESCRIPTION
**Pull request by moonyuet**
## Changelog Description
 Setting the timeline length based on DB record in 3dsMax Hosts
image

## Additional info
N/A

## Testing notes:
1. launch 3dsMax via Openpype
2. create new scene
3. check the timeline
4. locate "Set Frame Range" in openpype menu
5. click it
6. check the frame range or the timeline to see if it is same with the setting shown in DB

### CODE REVIEW

**Comment by BigRoy** [comment]
This PR seems to also include #4652 - is that correct?

**Comment by moonyuet** [comment]
> This PR seems to also include #4652 - is that correct?

yes, but they are in a separate ticket, I will wait the old one to merge and update this ticket.  I convert this to draft first

**Review by LiborBatek — APPROVED** [review]
All works as expected! Did several tests in multiple shots and all being ok.

image

File: openpype/hosts/max/api/lib.py [hunk 1]
```diff
@@ -157,6 +165,125 @@ def get_multipass_setting(project_setting=None):
...
+    handle_end = asset["data"].get("handleEnd")
+    if handle_end is None:
+        handle_end = handles
+
+    return {
+        "frameStart": frame_start,
+        "frameEnd": frame_end,
+        "handleStart": handle_start,
+        "handleEnd": handle_end
+    }
```
Comment (antirotor) [inline openpype/hosts/max/api/lib.py:240]: We should stop using `legacy_io`:

```suggestion
def get_frame_range() -> Dict:
    """Get the current assets frame range and handles.
    
    Returns:
    	Dict: with frame start, frame end, handle start, handle end.
    
    """
    # Set frame start/end
    asset = get_current_project_asset()

    frame_start = asset["data"].get("frameStart")
    frame_end = asset["data"].get("frameEnd")
    # Backwards compatibility
    if frame_start is None or frame_end is None:
        frame_start = asset["data"].get("edit_in")
        frame_end = asset["data"].get("edit_out")

    if frame_start is None or frame_end is None:
        return

    handles = asset["data"].get("handles") or 0
    handle_start = asset["data"].get("handleStart")
    if handle_start is None:
        handle_start = handles

    handle_end = asset["data"].get("handleEnd")
    if handle_end is None:
        handle_end = handles

    return {
        "frameStart": frame_start,
        "frameEnd": frame_end,
        "handleStart": handle_start,
        "handleEnd": handle_end
    }
```

File: openpype/hosts/max/api/lib.py [hunk 2]
```diff
@@ -157,6 +165,125 @@ def get_multipass_setting(project_setting=None):
...
+        rt.frameRate = fps_number
+
+    frame_range = get_frame_range()
+
+    frame_start = frame_range["frameStart"] - int(frame_range["handleStart"])
+    frame_end = frame_range["frameEnd"] + int(frame_range["handleEnd"])
+
+    frange_cmd = f"animationRange = interval {frame_start} {frame_end}"
+
+    rt.execute(frange_cmd)
```
Comment (antirotor) [inline openpype/hosts/max/api/lib.py:266]:
```suggestion
def reset_frame_range(fps: bool=True):
    """Set frame range to current asset.
    
    	This is part of 3dsmax documentation:

        animationRange: A System Global variable which lets you get and
        	set an Interval value that defines the start and end frames
        	of the Active Time Segment.
        frameRate: A System Global variable which lets you get
        	and set an Integer value that defines the current
        	scene frame rate in frames-per-second.
    """
    if fps:
        fps_number = float(get_current_project(fields=["data.fps"])["data"]["fps"])
        rt.frameRate = fps_number

    frame_range = get_frame_range()

    frame_start = frame_range["frameStart"] - int(frame_range["handleStart"])
    frame_end = frame_range["frameEnd"] + int(frame_range["handleEnd"])

    frange_cmd = f"animationRange = interval {frame_start} {frame_end}"

    rt.execute(frange_cmd)
```

File: openpype/hosts/max/api/lib.py [hunk 3]
```diff
@@ -157,6 +165,125 @@ def get_multipass_setting(project_setting=None):
...
+
+    Returns:
+        None
+
+    """
+    rt.renderWidth = width
+    rt.renderHeight = height
+
+
+def reset_scene_resolution():
```
Comment (antirotor) [inline openpype/hosts/max/api/lib.py:183]: Please see notes on #4652

File: openpype/hosts/max/api/lib.py [hunk 4]
```diff
@@ -157,6 +165,125 @@ def get_multipass_setting(project_setting=None):
                            ["multipass"])
 
 
+def set_scene_resolution(width, height):
```
Comment (antirotor) [inline openpype/hosts/max/api/lib.py:168]:
```suggestion
def set_scene_resolution(width: int, height: int):
```

**Review by antirotor — CHANGES_REQUESTED** [review]
`legacy_io` deprecation notes (similar to #4652)

**Review by antirotor — APPROVED** [review]

**Pull request merged**